File: front/src/services/desenvolvedorService.ts
import api from './api';
import type { Desenvolvedor } from '../types';

export const desenvolvedorService = {
    listar: async (): Promise<Desenvolvedor[]> => {
        const res = await api.get<Desenvolvedor[]>('/desenvolvedores');
        return res.data;
    },

    listarAtivos: async (): Promise<Desenvolvedor[]> => {
        const res = await api.get<Desenvolvedor[]>('/desenvolvedores/ativos');
        return res.data;
    },

    buscarPorId: async (id: number): Promise<Desenvolvedor> => {
        const res = await api.get<Desenvolvedor>(`/desenvolvedores/${id}`);
        return res.data;
    },

    criar: async (dto: Omit<Desenvolvedor, 'id'>): Promise<Desenvolvedor> => {
        const res = await api.post<Desenvolvedor>('/desenvolvedores', dto);
        return res.data;
    },

    atualizar: async (id: number, dto: Omit<Desenvolvedor, 'id'>): Promise<Desenvolvedor> => {
        const res = await api.put<Desenvolvedor>(`/desenvolvedores/${id}`, dto);
        return res.data;
    },

    deletar: async (id: number): Promise<void> => {
        await api.delete(`/desenvolvedores/${id}`);
    },
};

File: back/src/main/java/com/sustentacao/service/DesenvolvedorService.java
package com.sustentacao.service;

import com.sustentacao.dto.DesenvolvedorDTO;
import com.sustentacao.entity.Desenvolvedor;
import com.sustentacao.repository.DesenvolvedorRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class DesenvolvedorService {

    private final DesenvolvedorRepository repository;

    public List<DesenvolvedorDTO> listar() {
        return repository.findAll().stream().map(this::toDTO).collect(Collectors.toList());
    }

    public List<DesenvolvedorDTO> listarAtivos() {
        return repository.findByAtivoTrue().stream().map(this::toDTO).collect(Collectors.toList());
    }

    public DesenvolvedorDTO buscarPorId(Long id) {
        return repository.findById(id)
                .map(this::toDTO)
                .orElseThrow(() -> new RuntimeException("Desenvolvedor não encontrado: " + id));
    }

    public DesenvolvedorDTO salvar(DesenvolvedorDTO dto) {
        Desenvolvedor dev = new Desenvolvedor();
        dev.setNome(dto.getNome());
        dev.setEmail(dto.getEmail());
        dev.setAtivo(dto.getAtivo() != null ? dto.getAtivo() : true);
        return toDTO(repository.save(dev));
    }

    public DesenvolvedorDTO atualizar(Long id, DesenvolvedorDTO dto) {
        Desenvolvedor dev = repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Desenvolvedor não encontrado: " + id));
        dev.setNome(dto.getNome());
        dev.setEmail(dto.getEmail());
        if (dto.getAtivo() != null) {
            dev.setAtivo(dto.getAtivo());
        }
        return toDTO(repository.save(dev));
    }

    public void deletar(Long id) {
        repository.deleteById(id);
    }

    public DesenvolvedorDTO toDTO(Desenvolvedor dev) {
        DesenvolvedorDTO dto = new DesenvolvedorDTO();
        dto.setId(dev.getId());
        dto.setNome(dev.getNome());
        dto.setEmail(dev.getEmail());
        dto.setAtivo(dev.getAtivo());
        return dto;
    }
}
